MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift

```diff
@@ -24,39 +21,46 @@ final class MovieDetailPresenter {
     }
     
     func fetchAll() {
-        getMovieDetails()
-        getOverview()
+        //getMovieDetails()
+        //getOverview()
         getMostPopularCast()
         getCrew()
         getRecommendations()
         getReview()
     }
     
-    func getMovieDetails() {
-        useCase.getMovieDetails(for: movieId) { [weak self] result in
-            guard let self = self else { return }
-            
-            switch result {
-            case .success(let movie):
-                self.delegate?.fillDetailTitleView(with: DetailTitleViewModel(fromModel: movie))
-            case .failure(let error):
-                print("Loading error: \(error.localizedDescription)")
-            }
-        }
+//    func getMovieDetails() {
+//        useCase.getMovieDetails(for: movieId) { [weak self] result in
+//            guard let self = self else { return }
+//
+//            switch result {
+//            case .success(let movie):
+//                self.delegate?.fillDetailTitleView(with: DetailTitleViewModel(fromModel: movie))
+//            case .failure(let error):
+//                print("Loading error: \(error.localizedDescription)")
+//            }
+//        }
+//    }
+    var movieDetails: AnyPublisher<DetailTitleViewModel, Never> {
+        useCase
+            .getMovieDetails(for: movieId)
+            .map { DetailTitleViewModel(fromModel: $0) }
+            .receiveOnMain()
     }
     
-    func getOverview() {
-        useCase.getMovieOverview(for: movieId) { [weak self] result in
-            guard let self = self else { return }
-            
-            switch result {
-            case .success(let overview):
-                self.delegate?.fillOverview(with: overview)
-            case .failure(let error):
-                print("Loading error: \(error.localizedDescription)")
-            }
-        }
-    }
+//    func getOverview() {
+//        useCase.getMovieOverview(for: movieId) { [weak self] result in
+//            guard let self = self else { return }
+//
+//            switch result {
+//            case .success(let overview):
+//                self.delegate?.fillOverview(with: overview)
+//            case .failure(let error):
+//                print("Loading error: \(error.localizedDescription)")
+//            }
+//        }
+//    }
+
```

Comment:
no need for this

Reply:
[Deleted](https://github.com/lihter/Movie-App/pull/47/files#diff-d8a7df924093b6e29d479b7644fb3593ce7762a26b79b7f10f290852dfb0ce54L51-L62)